test/statsd/ecs_statsd_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package statsd
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+)
+
+var _ test_runner.IECSTestRunner = (*ECSStatsdTestRunner)(nil)
+
+const testRetryCount = 3
+
+type ECSStatsdTestRunner struct {
+	test_runner.ECSBaseTestRunner
+}
+
+func (t *ECSStatsdTestRunner) Validate(e *environment.MetaData) status.TestGroupResult {
+	metricsToFetch := t.GetMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		result := t.validateStatsdMetric(metricName)
+
+		// there seems to be some delay before the runner is able to fetch metrics from CW
```

Comment:
I wouldnt expect this, since the statsd container is basically emitting metrics in a loop from when it comes up. And the agent is run for a 3-min duration.
In addition, it seems there is already logic to account for some missing data points with the upper and lower bounds in the validate func below.

If we didnt need this in the EC2 test, worth checking why we need it here.

Reply:
Yeah, I'm not too clear exactly why there is some delay. I had a couple theories around it, but there shouldn't be any delays expected as you mentioned tnat statsd container keeps sending metrics regardless of the status of CWA.

Reply:
I suspect now that the delay is related to `portMappings`, and it's pretty much the same it takes to spin up ECS task with CWA. So I added sleep before starting the validation for this case.